### PR TITLE
Jennyf/g rpc

### PR DESCRIPTION
**Pull request by jennyf19**
work on template next

### CODE REVIEW

File: tests/WebAppCallsWebApiCallsGraph/Client/Controllers/HomeController.cs [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
+using System.Threading.Tasks;
 using WebApp_OpenIDConnect_DotNet.Models;
 
 namespace WebApp_OpenIDConnect_DotNet.Controllers
 {
     [Authorize]
     public class HomeController : Controller
     {
-        private readonly ITokenAcquisition _tokenAcquisition;
+        private ITokenAcquisition _tokenAcquisition;
```
Comment (jmprieur) [inline tests/WebAppCallsWebApiCallsGraph/Client/Controllers/HomeController.cs:19]: re-add the readonly?

File: tests/WebAppCallsWebApiCallsGraph/Client/appsettings.json [hunk 2]
```diff
@@ -29,6 +29,17 @@
...
 
     },
+    "SayHello": {
+        /*
+      TodoListScope is the scope of the Web API you want to call. This can be: "api://a4c2469b-cf84-4145-8f5f-cb7bacf814bc/access_as_user",
+      - a scope for a V2 application (for instance api://b3682cc7-8b30-4bd2-aaba-080c6bf0fd31/access_as_user)
+      - a scope corresponding to a V1 application (for instance <GUID>/user_impersonation, where  <GUID> is the
+        clientId of a V1 application, created in the https://portal.azure.com portal.
+    */
+        "Scopes": "api://a4c2469b-cf84-4145-8f5f-cb7bacf814bc/access_as_user", // "api://712ae8d7-548a-4306-95b6-ee9117ee86f0/access_as_user",
```
Comment (jmprieur) [inline tests/WebAppCallsWebApiCallsGraph/Client/appsettings.json:39]: Do we want to keep the comment? what was it?

File: tests/WebAppCallsWebApiCallsGraph/gRPC/Startup.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddGrpc();
+            services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
+                    .AddMicrosoftIdentityWebApi(Configuration, "AzureAd")
+                    .EnableTokenAcquisitionToCallDownstreamApi()
```
Comment (jmprieur) [inline tests/WebAppCallsWebApiCallsGraph/gRPC/Startup.cs:30]: We don't call a downstream API in SayHello. Do we want to keep these instructions?

**Review by jmprieur — APPROVED** [review]
Thanks @jennyf19 
Supporting this scenario is a great addition

I've left a few comments.

File: tests/WebAppCallsWebApiCallsGraph/Client/appsettings.json (hunk 2 above)
Comment (jennyf19) [reply tests/WebAppCallsWebApiCallsGraph/Client/appsettings.json:39]: which comment? i see them all there

File: tests/WebAppCallsWebApiCallsGraph/gRPC/Startup.cs (hunk 3 above)
Comment (jennyf19) [reply tests/WebAppCallsWebApiCallsGraph/gRPC/Startup.cs:30]: good point

**Pull request merged**